apps/prairielearn/src/pages/studentAssessmentInstance/studentAssessmentInstance.js

```diff
@@ -204,6 +238,20 @@ router.post(
         res.locals.authn_user.user_id,
       );
       res.redirect(req.originalUrl);
+    } else if (req.body.__action === 'delete_instance') {
+      if (!canDeleteAssessmentInstance(res.locals)) {
+        throw new error.HttpStatusError(403, 'Access denied');
+      }
+
+      await assessment.deleteAssessmentInstance(
+        res.locals.assessment.id,
+        res.locals.assessment_instance.id,
+        res.locals.authn_user.user_id,
+      );
+      flash('success', 'Assessment instance deleted.');
+      // TODO: decide whether to redirect back to the same assessment, or back
+      // to the assessments list.
+      res.redirect(`${res.locals.urlPrefix}/assessments`);
```

Comment:
Given the assessment list is can be fairly large, and usually my intention for removing my assessment instance is to make a new one, I'd vote for back to the same assessment.

Reply:
If we're going back to the same assessment, that'll pretty much always generate a new instance. Do we want to make the UI call this process "regenerate" instead of "delete"?

Reply:
Dave made this change in recent commits.